src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,150 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> prev;
+        Node<T> next = null;
+        prev = size == 0 ? null : tail;
+        Node<T> newNode = new Node<>(prev, value, next);
+        if (isEmpty()) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (size == 0 || index == size) {
+            if (index > size || index < 0) {
+                throw new IndexOutOfBoundsException("Index is incorrect!");
+            }
```

Comment:
move this check after this if block and make the message more informative (pass index value)